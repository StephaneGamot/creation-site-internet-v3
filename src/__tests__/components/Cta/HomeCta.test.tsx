import React from "react";
import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import Cta from "@/components/cta/HomeCta";

// ✅ Mock des traductions
jest.mock("next-intl", () => ({
  useTranslations: () => {
    return (key: string) => {
      if (key === "title") return "Une question ? Envie d’un site à votre image ?";
      if (key === "text") return "Discutons ensemble de votre projet web. Je suis à l’écoute.";
      if (key === "aria") return "Aller à la page de contact";
      if (key === "label") return "Me contacter";
      return "";
    };
  },
  useLocale: () => "fr",
}));

// ✅ Mock du composant ContactButton
jest.mock("@/components/Buttons/ContactButton", () => () => (
  <a href="/fr/contact" role="button">Me contacter</a>
));

describe("HomeCta", () => {
  it("affiche le titre et le texte", () => {
    render(<Cta />);
    expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent("Une question ? Envie d’un site à votre image ?");
    expect(screen.getByText("Discutons ensemble de votre projet web. Je suis à l’écoute.")).toBeInTheDocument();
  });

  it("contient le bouton Me contacter", () => {
    render(<Cta />);
    const button = screen.getByRole("button", { name: "Me contacter" });
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute("href", "/fr/contact");
  });

  it("passe les tests d’accessibilité", async () => {
    const { container } = render(<Cta />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it("correspond au snapshot", () => {
    const { asFragment } = render(<Cta />);
    expect(asFragment()).toMatchSnapshot();
  });
});
