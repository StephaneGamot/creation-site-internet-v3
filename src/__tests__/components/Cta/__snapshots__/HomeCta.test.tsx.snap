// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HomeCta correspond au snapshot 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="cta-title"
    class="text-center my-20"
  >
    <h2
      class="text-3xl font-title font-semibold mb-4"
      id="cta-title"
    >
      Une question ? Envie d’un site à votre image ?
    </h2>
    <p
      class="text-lg text-gray-700 mb-6"
    >
      Discutons ensemble de votre projet web. Je suis à l’écoute.
    </p>
    <a
      href="/fr/contact"
      role="button"
    >
      Me contacter
    </a>
  </section>
</DocumentFragment>
`;
